slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbHealthCheck.java

```diff
@@ -0,0 +1,75 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.slashingprotection;
+
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class DbHealthCheck implements Runnable {
+
+  private final Optional<SlashingProtectionContext> slashingProtectionContext;
+  private final int dbHealthCheckTimeoutMilliseconds;
+  private final int dbHealthCheckIntervalMilliseconds;
+  private final AtomicBoolean isDbDown;
+
+  public DbHealthCheck(
+      final Optional<SlashingProtectionContext> slashingProtectionContext,
+      final int dbHealthCheckTimeoutMilliseconds,
+      final int dbHealthCheckIntervalMilliseconds) {
+    this.slashingProtectionContext = slashingProtectionContext;
+    this.dbHealthCheckTimeoutMilliseconds = dbHealthCheckTimeoutMilliseconds;
+    this.dbHealthCheckIntervalMilliseconds = dbHealthCheckIntervalMilliseconds;
+    this.isDbDown = new AtomicBoolean(false);
+  }
+
+  @Override
+  public void run() {
+    slashingProtectionContext.ifPresent(
+        protectionContext ->
+            Executors.newScheduledThreadPool(1)
+                .scheduleAtFixedRate(
+                    () -> {
+                      Future<Integer> future =
+                          Executors.newCachedThreadPool()
+                              .submit(
+                                  () ->
+                                      protectionContext
+                                          .getSlashingProtectionJdbi()
+                                          .withHandle(
+                                              h ->
+                                                  h.createQuery("SELECT 1")
+                                                      .mapTo(Integer.class)
+                                                      .one()));
+                      try {
+                        // Check db health with timeout.
+                        future.get(this.dbHealthCheckTimeoutMilliseconds, TimeUnit.MILLISECONDS);
+                        isDbDown.set(false);
+                      } catch (Exception e) {
+                        // Have exception in database health check (timeout or error).
+                        isDbDown.set(true);
+                      } finally {
+                        future.cancel(true);
+                      }
+                    },
+                    this.dbHealthCheckIntervalMilliseconds,
+                    this.dbHealthCheckIntervalMilliseconds,
+                    TimeUnit.MILLISECONDS));
+  }
+
+  public boolean isDbDown() {
```

Comment:
Would prefer this was written in the positive to make the code clearer i.e. isDbUp